background/transport.js

```diff
@@ -53,10 +70,49 @@ export default class Transport {
     return this.secureApiMessage(port);
   }
 
+  setIcon (status) {
+    console.log('setting icon to ', status);
+    const ctx = document.createElement('canvas').getContext('2d');
+    const image = new Image();
+
+    const size = 76;
+
+    image.onload = () => {
+      ctx.drawImage(image, 0, 0);
+
+      switch (status) {
+        case 'connected':
+          ctx.fillStyle = 'rgba(46, 204, 113, 0.5)';
+          break;
+
+        case 'disconnected':
+        default:
+          ctx.fillStyle = 'rgba(231, 76, 60, 0.25)';
+          break;
+      }
+
+      ctx.strokeStyle = 'white';
+      ctx.beginPath();
```

Comment:
Have you tried drawing it in upper left corner? I think now it looks a bit crowdy when badge is displayed as well.